packages/ember-model/lib/belongs_to.js

```diff
@@ -96,7 +96,9 @@ Ember.belongsTo = function(type, options) {
 
       if (meta.options.embedded) {
         if (oldValue) {
-          oldValue.removeObserver('isDirty', dirtyChanged);
+          try {
+            oldValue.removeObserver('isDirty', dirtyChanged);
+          } catch(e) { }
```

Comment:
The `removeObserver` sometimes fails with a `Assertion Failed: You attempted to remove a function listener which did not exist on the instance, which means you may have attempted to remove it before it was added` message.

Adding a [`hasObserverFor`](https://github.com/emberjs/ember.js/blob/4a98e1610b795edb544513f10a8870af1375141d/packages/%40ember/-internals/runtime/lib/mixins/observable.js#L401) guard doesn't seem to prevent this from raising exceptions.

This message was changed from a `deprecate` to an `assert` for Ember 3.11. Given that we're almost off ember-model, I think swallowing the exception is the pragmatic thing to do in the short term given that essentially what's currently happening in our production app.